noir-projects/noir-contracts/contracts/easy_private_voting_contract/src/main.nr

```diff
@@ -21,14 +22,26 @@ contract EasyPrivateVoting {
     fn constructor(admin: AztecAddress) {
         storage.admin.write(admin);
         storage.vote_ended.write(false);
+        storage.active_at_block.initialize(context.historical_header.global_variables.block_number as u32);
     }
     // docs:end:constructor
 
     // docs:start:cast_vote
     #[aztec(private)] // annotation to mark function as private and expose private context
     fn cast_vote(candidate: Field) {
-        let msg_sender_npk_m_hash = get_npk_m_hash(&mut context, context.msg_sender());
-        // TODO (#6312): This will break with key rotation. Fix this. Can vote multiple times by rotating keys.
+        let active_at_block = storage.active_at_block.read_private();
+        assert(
```

Comment:
I'm not sure that you would could end up in this one failing? Namely, to call this function you have that the contract must be deployed and since it must be initialized before letting you call here think we cannot hit it? 🤔

Regardless, I think you will already have this handled from the `get_header_at`? 🤔